sdks/java/core/src/main/java/org/apache/beam/sdk/io/DefaultFilenamePolicy.java

```diff
@@ -95,36 +149,67 @@ public static DefaultFilenamePolicy constructUsingStandardParameters(
    * strings.
    *
    * <p>Within a shard template, repeating sequences of the letters "S" or "N"
-   * are replaced with the shard number, or number of shards respectively.  The
-   * numbers are formatted with leading zeros to match the length of the
+   * are replaced with the shard number, or number of shards respectively.
+   * Repeating sequence of "P" is replaced with the index of the window pane.
+   * "L" is replaced with "true" in case of last pane or "false" otherwise.
+   * "F" is replaced with "true" in case of first pane or "false" otherwise.
```

Comment:
ok.. so then only W and P are needed... and W will be replaced by nice window-to-string and P will be replaced with nice pane-to-string

right?